### DISCUSSION TITLE
Resolve text in custom block with blocks resolver

### DISCUSSION THREAD

**Discussion by alancwoo** [kirby-tools/community#26] [Q&A] [answered]
Say I have a custom block defined like:

```
fields:
  description:
    type: blocks
    fieldsets:
      - type: test-custom
           name: Test Custom
           fields:
             excerpt:
               type: textarea
```

If i want to adjust this block's output via a resolver, is it possible via:

```
// config.php
return [
    'blocksResolver' => [
        'resolvers' => [
            'test-custom:excerpt' => fn (Field $field, Block $block) => [
                // adjust output
            ]
        ]
    ]
];
```

I can't seem to catch this `block:field` output at all, although I'm able to resolve `file` fields within my custom block.

```
getkirby/cms: 4.5.0
johannschopplich/kirby-headless: 4.0.2
```

**Comment by johannschopplich** [comment]
Your code example looks fine. Could you please provide a minimum reproducible example, based on either the Kirby plainkit or starterkit? Thanks in advance!

**Reply by alancwoo** [reply]
Thanks for the quick response, I've set up a minimal reproduction here based on plainkit: 
[headless-test.zip](https://github.com/user-attachments/files/18063404/headless-test.zip)

The `home` page has a custom block and you can see in the `test-custom` block, `thumbnail` field resolves, but I can't seem to adjust the `excerpt`.

Relevant files:
- `/site/config/config.php`
- `/site/blueprint/pages/default.yml`
- `/site/templates/default.php`

**Reply by johannschopplich** [reply, marked as answer]
I found the problem: Your config was not correct. You have to use the [`resolvers` options](https://kirby.tools/docs/headless/field-methods#custom-resolvers) to define custom resolvers. Feel free to dig into the docs to learn more about them.

**tl;dr:**

```php
"blocksResolver" => [
    // See https://kirby.tools/docs/headless/field-methods
    "resolvers" => [
        "test-custom:excerpt" => function ( $field,  $block) {
            return "Hello World";
        },
    ]
]
```
